### PR TITLE
libzzip: update to 0.13.72

### PR DESCRIPTION
**Pull request by catap**
#### Description

This pull request really updated libzzip to the last version.

It also switched to using `cmake` build which allow to overstep an issue with caching `Cflags` and `Libs` from build and injects it to anyone who used it.

Just try to install `libzzip` before this PR and run:

```sh
cat /opt/local/lib/pkgconfig/zzip*
```

If your sdk haven't match with cached one and application used this lib, it injects two `-isysroot` which creates quite an issue.

Good example of such case is: https://github.com/macports/macports-ports/pull/12222

This pull request depends on:
 - https://github.com/macports/macports-ports/pull/12241
 - https://github.com/macports/macports-ports/pull/12270
 - https://github.com/macports/macports-ports/pull/12281
 - https://github.com/macports/macports-ports/pull/12320

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.5.2 20G95 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@reneeotten here an root cause of issue with MPD.

Don't ask me how I found it.

File: archivers/libzzip/Portfile
Comment (reneeotten) [inline archivers/libzzip/Portfile:37]: is just doing `${prefix}/bin/python${python.branch}` not sufficient? That would avoid the need to define `python.prefix` and `python.bin`

File: archivers/libzzip/Portfile
Comment (reneeotten) [inline archivers/libzzip/Portfile:71]: here you can use `configure.args replace` instead

File: archivers/libzzip/Portfile
Comment (reneeotten) [inline archivers/libzzip/Portfile:77]: can we keep the tests enabled?

**Comment by reneeotten** [comment]
please note that there was already a PR open for this update: #11760. Can you please check that PR and either comment there so that your changes (if different) can be incorporated there or incorporate them here?

**Comment by catap** [comment]
@reneeotten about another PR. You're right. I'll incorporated some changes because this one looks forgotten.

File: archivers/libzzip/Portfile
Comment (catap) [reply archivers/libzzip/Portfile:77]: @reneeotten well... with migration to `cmake` they run tests during the build, and it failed with a error:
```
dyld: Library not loaded: /opt/local/lib/libzzip.13.dylib
  Referenced from: /opt/local/var/macports/build/_Users_catap_src_macports-ports_archivers_libzzip/libzzip/work/build/test/./zzipsetstub
  Reason: image not found
```

Can I fix it? Prehabs. Bug I feel it as waste of time. A lot of time.

**Comment by reneeotten** [comment]
I'll leave this one for @mojca to decide whether this or the previous PR has reached a point where it can be merged.

**Comment by catap** [comment]
@reneeotten / @mojca I've almost fixed tests.

Anyway, one test is failed and I've opened an upstream issue: https://github.com/gdraheim/zziplib/issues/120

File: archivers/libzzip/Portfile
Comment (mojca) [inline archivers/libzzip/Portfile:35]: How does this work without the patch that was provided in the other PR?

Comment (catap) [reply archivers/libzzip/Portfile:35]: @mojca by suggesting cmake which version of python it should use. I just bypass all cmake's attempts to find it.

The good idea is adding `-DPYTHON_EXECUTABLE=${prefix}/bin/python${python.branch}` to `cmake` PortGroup but I'm too lazy to do it :)

**Comment by catap** [comment]
@mojca I've added link for both issues which I've opened during this work and removed `DYLD_LIBRARY_PATH` hacks which doesn't required right now.

**Comment by catap** [comment]
@mojca and I've fixed the tests :)

File: archivers/libzzip/Portfile
Comment (catap) [reply archivers/libzzip/Portfile:77]: Tests are fixed.

**Comment by reneeotten** [comment]
@macportsbot retry

**Comment by catap** [comment]
finaly, this should pass CI and as soon as it merged it is possible to update mpd via https://github.com/macports/macports-ports/pull/12222

**Comment by catap** [comment]
@reneeotten ping here!

**Comment by catap** [comment]
rebased it to last master

**Comment by pmetzger** [comment]
No idea, I haven't been following this. Do you believe that things are ready to be merged? Do @mojca and @ryandesign agree?

**Comment by catap** [comment]
@pmetzger yes, I do. As soon as it merged, it unblocks `mpd` and probably something else.

**Comment by pmetzger** [comment]
@mojca can this be merged?

**Comment by catap** [comment]
@pmetzger I guess this can be treated as yes by abandoned protocol :)

**Comment by pmetzger** [comment]
I just noticed this bumps texlivebin and cmconvert, can you explain the rationale?

**Comment by catap** [comment]
@pmetzger all of them depends on this library.

ABI and library name is changed => it need to be rebuilded

**Comment by pmetzger** [comment]
I'd still feel more comfortable with @mojca or @ryandesign commenting before merge.

**Comment by catap** [comment]
@ryandesign ping here

**Comment by pmetzger** [comment]
Failing any objections from @ryandesign or @mojca I will merge this tomorrow.

**Comment by ryandesign** [comment]
Have nothing further to add.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
> Failing any objections from @ryandesign or @mojca I will merge this tomorrow.

Sounds good, let's give @mojca a chance to chime in today.

**Comment by pmetzger** [comment]
@mascguy Feel free to merge this tomorrow if I don't get to it.

**Pull request merged**
